onlinejudge/service/atcoder.py

```diff
@@ -197,7 +197,7 @@ def from_url(cls, url: str) -> Optional['AtCoderContest']:
 
         # example: https://atcoder.jp/contests/agc030
         if result.scheme in ('', 'http', 'https') and result.hostname in ('atcoder.jp', 'beta.atcoder.jp'):
-            m = re.match(r'^/contests/([\w\-_]+)/?$', utils.normpath(result.path))
+            m = re.match(r'^/contests/([\w\-_]+)/?$.*', utils.normpath(result.path))
```

Comment:
```suggestion
            m = re.match(r'/contests/([\w\-_]+)/?.*', utils.normpath(result.path))
```

`$` は文字列の末尾にマッチする記号なので、 `$.*` でなく `.*$` が適切です。
(加えて、 `re.match` は文字列の全体にマッチする (つまり、常に前後に `^` と `$` を付ける) ような関数なので、これらを取ってしまうのがよいでしょう。これは元から存在していたミスです)